tensorboard/plugins/mesh/polymer3/tf_mesh_dashboard/tf-mesh-dashboard.ts

```diff
@@ -209,123 +205,123 @@ <h3>No point cloud data was found.</h3>
-              Object.assign({}, item, {
-                sample: i,
-                ofSamples: samples,
-              })
-            );
-          }
-          const withSamples = baseCategories.map((category) =>
-            Object.assign({}, category, {
-              items: [].concat.apply([], category.items.map(explodeItem)),
-            })
-          );
-          return withSamples;
-        },
-      }); // End of Polymer constructor call.
-    })(); // End of anonymous namespace.
-  </script>
-</dom-module>
+  `;
+
+  @property({type: Boolean})
+  reloadOnReady: boolean = true;
+
+  @property({type: Array})
+  _selectedRuns: string[];
+
+  @property({type: Object})
+  _runToTagInfo: object;
+
+  @property({type: Boolean})
+  _dataNotFound: boolean;
+
+  @property({type: String})
+  _tagFilter: string = '.*';
+
+  @property({
+    type: String,
+    notify: true,
+  })
+  _selectedView: string = 'all';
+
+  @property({type: Object})
+  _requestManager = new RequestManager();
+
+  constructor() {
+    super();
+    window.addEventListener(
+      'resize',
+      () => {
+        this._handleWindowResize();
+      },
+      false
+    );
+    if (this.reloadOnReady) this.reload();
+  }
+
+  _getAllChildren() {
+    return Array.from(
+      this.shadowRoot.querySelectorAll('tf-mesh-loader')
+    ) as TfMeshLoader[];
+  }
+
+  _onCameraPositionChanged(event) {
+    if (this._selectedView == 'share') {
+      this._getAllChildren().forEach((g) => {
+        if (event.target == g) return; // Do not update trigger camera.
+        (g as any).setCameraViewpoint(
+          event.detail.position,
+          event.detail.far,
+          event.detail.target
+        );
+      });
+    }
+  }
+
+  _shouldOpen(index) {
+    return index <= 2;
+  }
+
+  reload() {
+    this._fetchTags().then(this._reloadMeshes.bind(this));
+  }
+
+  _handleWindowResize() {
+    this._getAllChildren().forEach((g) => {
+      (g as any).redraw();
+    });
+  }
+
+  _fetchTags() {
+    const url = getRouter().pluginRoute('mesh', '/tags');
+    return this._requestManager.request(url).then((runToTagInfo) => {
+      if (_.isEqual(runToTagInfo, this._runToTagInfo)) {
+        // No need to update anything if there are no changes.
+        return;
+      }
+      const runToTag = _.mapValues(runToTagInfo, (x) => Object.keys(x));
+      const tags = getTags(runToTag);
+      this._dataNotFound = tags.length === 0;
+      this._runToTagInfo = runToTagInfo;
+    });
+  }
+  _reloadMeshes() {
+    this._getAllChildren().forEach((g) => {
+      g.reload();
+    });
+  }
+
+  @computed('_runToTagInfo', '_selectedRuns', '_tagFilter')
+  get _categories(): unknown[] {
+    var runToTagInfo = this._runToTagInfo;
+    var selectedRuns = this._selectedRuns;
+    var tagFilter = this._tagFilter;
+    const runToTag = _.mapValues(runToTagInfo, (x) => Object.keys(x));
+    const baseCategories = categorizeRunTagCombinations(
+      runToTag as any,
```

Comment:
FWIW, this could be `runToTag as tf_categorization_utils.RunToTag` if we want to avoid the extra `as any`.

Reply:
Done.